src/image/loading_displaying.js

```diff
@@ -158,6 +159,341 @@ p5.prototype.loadImage = function(path, successCallback, failureCallback) {
   return pImg;
 };
 
+/**
+ * Generates a gif of your current animation and downloads it to your computer!
+ *
+ * The duration argument specifies how many seconds you want to record from your animation.
+ * This value is then converted to the necessary number of frames to generate it.
+ *
+ * With the delay argument, you can tell the function to skip the first `delay` seconds
+ * of the animation, and then download the `duration` next seconds. This means that regardless
+ * of the value of `delay`, your gif will always be `duration` seconds long.
+ *
+ * @method saveGif
+ * @param  {String} filename File name of your gif
+ * @param  {Number} duration Duration in seconds that you wish to capture from your sketch
+ * @param  {Number} delay Duration in seconds that you wish to wait before starting to capture
+ *
+ * @example
+ * <div>
+ * <code>
+ * function setup() {
+ *   createCanvas(100, 100);
+ *   colorMode(HSL);
+ * }
+
+ * function draw() {
+ *   // create some cool dynamic background
+ *   let hue = map(sin(frameCount / 100), -1, 1, 0, 100);
+ *   background(hue, 40, 60);
+
+ *   // create a circle that moves diagonally
+ *   circle(
+ *     100 * sin(frameCount / 10) + width / 2,
+ *     100 * sin(frameCount / 10) + height / 2,
+ *     10
+ *   );
+ * }
+
+ * // you can put it in the mousePressed function,
+ * // or keyPressed for example
+ * function mousePressed() {
+ *   // this will download the first two seconds of my animation!
+ *   saveGif('mySketch', 2);
+ * }
+ * </code>
+ * </div>
+ *
+ * @alt
+ * animation of a circle moving smoothly diagonally
+ */
+p5.prototype.saveGif = async function(fileName, seconds = 3, delay = 0) {
+  // validate parameters
+  p5._validateParameters('saveGif', arguments);
+  // get the project's framerate
+  // if it is undefined or some non useful value, assume it's 60
+  let _frameRate = this._targetFrameRate;
+  if (_frameRate === Infinity || _frameRate === undefined || _frameRate === 0) {
+    _frameRate = 60;
+  }
+
+  // calculate delay based on frameRate
+  let gifFrameDelay = 1 / _frameRate * 1000;
+
+  // constrain it to be always greater than 20,
+  // otherwise it won't work in some browsers and systems
+  // reference: https://stackoverflow.com/questions/64473278/gif-frame-duration-seems-slower-than-expected
+  gifFrameDelay = gifFrameDelay < 20 ? 20 : gifFrameDelay;
+
+  // because the input was in seconds, we now calculate
+  // how many frames those seconds translate to
+  let nFrames = Math.ceil(seconds * _frameRate);
```

Comment:
Just because I thought it was a bit more convenient to have some extra frames (because ceil rounded to higher integer). If you have extra unnecessary frames, you can drop them down easily, but if you missed the last frames, you need to re-render the whole thing again.

Does it make sense? Or is it better to floor it?

Reply:
given the main focus is on the gif and its appearence, i think an extra frame wont be producing much glitch.
we can go with it.